### DISCUSSION TITLE
Error installing shadcn-ui on my local machine

### DISCUSSION THREAD

**Discussion by Chukwu3meka** [shadcn-ui/ui#6027] [Q&A]
I've been able to clone locally on my windows machine, but running `pnpm install` is failing at the point of installing `sharp`

Initially I thought it had to do with pnpm, so I deleted /node_modules and then run `pnpm store prune`. Packages were added `Progress: resolved 1419, reused 0, downloaded 1409, added 1419, done` but `Running install script, failed in 5.9s`. 

Could it have to do with proxy settings since I'm getting this `sharp: Installation error: Request timed out`, and the terminal download site for sharp is not reachable when I try to open it from the browser ` https://github.com/lovell/sharp-libvips/releases/download/v8.13.3/libvips-8.13.3-win32-x64.tar.br`, is it windows OS or something else I'm not aware of?

![image](https://github.com/user-attachments/assets/9fd935a4-2b2d-47d4-959a-ec678c5cf454)

`
Scope: all 4 workspace projects
Lockfile is up to date, resolution step is skipped
Packages: +1419
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Downloading @img/sharp-win32-x64@0.33.4: 8.65 MB/8.65 MB, done
Downloading next@14.3.0-canary.43: 23.75 MB/23.75 MB, done
Downloading @next/swc-win32-x64-msvc@14.3.0-canary.43: 51.13 MB/51.13 MB, done
Downloading imagescript@1.3.0: 10.69 MB/10.69 MB, done
Downloading typescript@4.9.5: 11.62 MB/11.62 MB, done
Downloading turbo-windows-64@1.9.9: 20.56 MB/20.56 MB, done
Progress: resolved 1419, reused 0, downloaded 1409, added 1419, done
node_modules/.pnpm/esbuild@0.21.5/node_modules/esbuild: Running postinstall script, done in 5.5s
node_modules/.pnpm/esbuild@0.18.20/node_modules/esbuild: Running postinstall script, done in 5.1s
node_modules/.pnpm/esbuild@0.17.19/node_modules/esbuild: Running postinstall script, done in 5.1s
node_modules/.pnpm/puppeteer@23.6.0_typescript@5.5.3/node_modules/puppeteer: Running postinstall script, done in 1.9s
node_modules/.pnpm/turbo@1.9.9/node_modules/turbo: Running postinstall script, done in 8.4s
node_modules/.pnpm/sharp@0.33.4/node_modules/sharp: Running install script, done in 92ms
node_modules/.pnpm/protobufjs@7.3.0/node_modules/protobufjs: Running postinstall script, done in 70ms
node_modules/.pnpm/sharp@0.31.3/node_modules/sharp: Running install script, failed in 5.9s
.../sharp@0.31.3/node_modules/sharp install$ (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
│ sharp: Downloading https://github.com/lovell/sharp-libvips/releases/download/v8.13.3/libvips-8.13.3-win32-x64.tar.br
│ sharp: Installation error: Request timed out
│ sharp: Please see https://sharp.pixelplumbing.com/install for required dependencies
`

**Comment by georgeeburt** [comment]
This is more than likely an issue related to your proxy settings, yes.

You can configure your network/proxy settings accordingly with pnpm:

```
pnpm config set proxy http://your-proxy-url:port
pnpm config set https-proxy http://your-proxy-url:port
```

**Reply by Chukwu3meka** [reply]
After updating proxy config, I'm getting this error `Installation error: unable to get local issuer certificate`

![image](https://github.com/user-attachments/assets/914d7267-dc8f-4631-824d-ccab19fbbded)

**Reply by georgeeburt** [reply]
This tells me that your proxy is using its own self-signed certificates? 

You need to obtain your proxy's certificate (usually available from your network administrator), then add it to your trusted certificates store:

For windows, use the _Certificates Management Console_ (`certlm.msc`) and import the certificate into "Trusted Root Certification Authorities".

If the system's certificate store is not enough, you can directly configure Node.js to use the certificate:

`set NODE_EXTRA_CA_CERTS=C:\path\to\proxy-cert.pem`

**Reply by Chukwu3meka** [reply]
Thanks, tried on a different machine, working fine without proxy issues
